CRM/Financial/BAO/ExportFormat/DataProvider/Sage50CSVProvider.php

```diff
@@ -0,0 +1,176 @@
+      ft.trxn_id AS trxn_id,
+      CASE
+        WHEN efti.entity_id IS NOT NULL
+        THEN efti.amount
+        ELSE eftc.amount
+      END AS amount,
+      fa_from.account_type_code AS credit_account_type_code,
+      fa_from.accounting_code AS credit_account,
+      fa_from.name AS credit_account_name,
+      fac.account_type_code AS from_credit_account_type_code,
+      fac.accounting_code AS from_credit_account,
+      fac.name AS from_credit_account_name,
+      fi.amount as net_amount,
+      eftc.id as civicrm_entity_financial_trxn_id,
+      li.label as item_description,
+      li.tax_amount as line_item_tax_amount,
+      li.financial_type_id as line_item_financial_type_id
+    FROM civicrm_entity_batch eb
+             LEFT JOIN civicrm_financial_trxn ft ON (eb.entity_id = ft.id AND eb.entity_table = 'civicrm_financial_trxn')
+             LEFT JOIN civicrm_financial_account fa_from ON fa_from.id = ft.from_financial_account_id
+             LEFT JOIN civicrm_entity_financial_trxn eftc ON (eftc.financial_trxn_id  = ft.id AND eftc.entity_table = 'civicrm_contribution')
+             LEFT JOIN civicrm_contribution c ON c.id = eftc.entity_id
+             LEFT JOIN civicrm_contact con on con.id = c.contact_id
+             LEFT JOIN civicrm_entity_financial_trxn efti ON (efti.financial_trxn_id  = ft.id AND efti.entity_table = 'civicrm_financial_item')
+             LEFT JOIN civicrm_financial_item fi ON fi.id = efti.entity_id
+             LEFT JOIN civicrm_line_item li ON (li.id = fi.entity_id AND fi.entity_table = 'civicrm_line_item')
+             LEFT JOIN civicrm_financial_account fac ON fac.id = fi.financial_account_id
+    WHERE eb.batch_id = ( %1 )";
+
+    CRM_Utils_Hook::batchQuery($sql);
+
+    $params = [1 => [$batchId, 'Integer']];
+    $dao = CRM_Core_DAO::executeQuery($sql, $params);
+
+    return $dao;
+  }
+
+  /**
+   * @param $exportResultDao
+   *
+   * @return array
+   */
+  public function formatDataRows($exportResultDao) {
+    $financialItems = $queryResults = [];
+    while ($exportResultDao->fetch()) {
+      $item = [
+        self::TYPE_LABEL => NULL,
+        self::ACCOUNT_REFERENCE_LABEL => 'CiviCRM',
+        self::NOMINAL_AC_REF_LABEL => NULL,
+        self::DEPARTMENT_CODE_LABEL => NULL,
+        self::DATE_LABEL => CRM_Utils_Date::customFormat($exportResultDao->trxn_date, '%d/%m/%Y'),
+        self::REFERENCE_LABEL => $exportResultDao->invoice_number,
+        self::DETAILS_LABEL => NULL,
+        self::NET_AMOUNT_LABEL => NULL,
+        self::TAX_CODE_LABEL => NULL,
+        self::TAX_AMOUNT_LABEL => $exportResultDao->line_item_tax_amount,
+        self::EXCHANGE_RATE_LABEL => NULL,
+        self::EXTRA_REFERENCE => NULL,
+        self::USER_NAME_LABEL => NULL,
+        self::PROJECT_REFN_LABEL => NULL,
+        self::COST_CODE_REFN_LABEL => NULL,
+      ];
+
+      if ($exportResultDao->is_payment == 0) {
+        $formattedItem = self::formatFinancialItemLines($item, $exportResultDao);
```

Comment:
@erawat, since the function is not static, I think it's better to call it from the current instance.